source/nodejs/adaptivecards-designer/src/app.ts

```diff
@@ -578,43 +655,58 @@ class Splitter {
         this._splitterElement.onpointerdown = (e: PointerEvent) => { this.pointerDown(e); };
         this._splitterElement.onpointermove = (e: PointerEvent) => { this.pointerMove(e); };
         this._splitterElement.onpointerup = (e: PointerEvent) => { this.pointerUp(e); };
-    }    
+    }
 }
 
 var app: DesignerApp;
 var horizontalSplitter: Splitter;
-var verticalSplitter: Splitter;
+var propertyVerticalSplitter: Splitter;
+var treeViewVerticalSplitter: Splitter;
 
 window.onload = () => {
-    document.getElementById("btnNewCard").onclick = (e) => {
-        app.newCard();
-    }
+    const fullScreenHandler = new FullScreenHandler(document.querySelector(".js-enter-fullscreen"));
+    fullScreenHandler.init();
+
+    new Clipboard(".js-copy-json", {
+        text: function () {
+            return JSON.stringify(app.card.toJSON(), null, 4);
+        }
+    });
 
     horizontalSplitter = new Splitter(document.getElementById("horizontalSplitter"), document.getElementById("jsonEditorHost"));
     horizontalSplitter.onRezized = (splitter: Splitter) => {
         if (isMonacoEditorLoaded) {
             monacoEditor.layout();
-        }    
+        }
     }
 
-    verticalSplitter = new Splitter(document.getElementById("verticalSplitter"), document.getElementById("propertySheetHost"));
-    verticalSplitter.isVertical = true;
-    verticalSplitter.onRezized = (splitter: Splitter) => {
+    propertyVerticalSplitter
+        = new Splitter(document.getElementById("propertyVerticalSplitter"), document.getElementById("propertySheetHost"));
+    propertyVerticalSplitter.isVertical = true;
+    propertyVerticalSplitter.onRezized = (splitter: Splitter) => {
         scheduleLayoutUpdate();
     }
 
+    treeViewVerticalSplitter
+        = new Splitter(document.getElementById("treeViewVerticalSplitter"), document.getElementById("treeViewSheetHost"));
+    treeViewVerticalSplitter.isVertical = true;
+
     let card = new Adaptive.AdaptiveCard();
     card.onImageLoaded = (image: Adaptive.Image) => {
         scheduleLayoutUpdate();
     }
 
     app = new DesignerApp(document.getElementById("designerHost"));
     app.propertySheetHostElement = document.getElementById("propertySheetHost");
+    app.treeViewSheetHostElement = document.getElementById("treeViewSheetHost");
     app.commandListHostElement = document.getElementById("commandsHost");
-    app.paletteHostElement = document.getElementById("toolPalette");
+    app.paletteHostElement = document.querySelector(".aside-items");
```

Comment:
Why querySelector rather than getElementById? getElementById is more appropriate given that it can only map to a single element, whereas querySelector returns the first element found that matches the selector...
This comment might apply to other places.

Reply:
P0: getElementById is safer and more solid

Reply:
In progress.